kotlin-checks-test-sources/src/main/kotlin/checks/RedundantMethodsInDataClassesCheckSample.kt

```diff
@@ -0,0 +1,104 @@
+package checks
+
+import java.util.Objects
+
+class RedundantMethodsInDataClassesCheckSample {
+
+    data class Person1(val name: String, val age: Int) {
+        override fun equals(other: Any?): Boolean { // Noncompliant {{Remove this redundant method which is the same as a default one.}}
+//                   ^^^^^^
+            return other is Person1 && other.name == name && other.age == age
+        }
+
+        override fun hashCode() = Objects.hash(name, age) // Noncompliant {{Remove this redundant method which is the same as a default one.}}
```

Comment:
Actually, if you look at the bytecode, it's not what is actually generated. 

```
// access flags 0x1
  public hashCode()I
    ALOAD 0
    GETFIELD checks/RedundantMethodsInDataClassesCheckSample$Person2.name : Ljava/lang/String;
    DUP
    IFNULL L0
    INVOKEVIRTUAL java/lang/Object.hashCode ()I
    GOTO L1
   L0
    POP
    ICONST_0
   L1
    BIPUSH 31
    IMUL
    ALOAD 0
    GETFIELD checks/RedundantMethodsInDataClassesCheckSample$Person2.age : I
    INVOKESTATIC java/lang/Integer.hashCode (I)I
    IADD
    IRETURN
    MAXSTACK = 2
    MAXLOCALS = 1
```

it should be something like: `31 * name.hashCode() + Integer.hashCode(age)`

Under the hood `Objects.hash()` calls `Arrays.hashCode` which is implemented with the same algorithm. So basically both function calls should be noncompliant.

Reply:
You can find out more information about the default implementations here: https://www.baeldung.com/kotlin/data-class-equals-method#2-understanding-data-classs-equals-method-and-solving-the-problem